npm/qsharp/generate_samples_content.js

```diff
@@ -7,26 +7,52 @@ import { readFileSync, writeFileSync } from "node:fs";
 import { dirname, join } from "node:path";
 import { fileURLToPath } from "node:url";
 
-import samples from "../../samples/samples.mjs";
+import qSharpSampleList from "../../samples/samples.mjs";
+import openQasmSampleList from "../../samples/OpenQASM/samples.mjs";
 
 const thisDir = dirname(fileURLToPath(import.meta.url));
-const sampleDir = join(thisDir, "..", "..", "samples");
-const sampleGeneratedDir = join(thisDir, "src");
-
-const result = samples.map((sample) => {
-  const samplePath = join(sampleDir, sample.file);
-  const sampleText = readFileSync(samplePath, "utf8");
-  return {
-    title: sample.title,
-    shots: sample.shots,
-    code: sampleText,
-    omitFromTests: sample.omitFromTests,
-  };
-});
-
-const contentPath = join(sampleGeneratedDir, "samples.generated.ts");
-writeFileSync(
-  contentPath,
-  `export default ${JSON.stringify(result, undefined, 2)}`,
-  "utf-8",
+const qSharpSampleDir = join(thisDir, "..", "..", "samples");
+const openQasmSampleDir = join(thisDir, "..", "..", "samples", "OpenQASM");
+
+const tsDir = join(thisDir, "src");
+const qSharpGeneratedTsPath = join(tsDir, "samples.generated.ts");
+const openQasmGeneratedTsPath = join(tsDir, "openqasm-samples.generated.ts");
+
+embedSampleContentsInTsFile(
+  qSharpSampleList,
+  qSharpSampleDir,
+  qSharpGeneratedTsPath,
+);
+embedSampleContentsInTsFile(
+  openQasmSampleList,
+  openQasmSampleDir,
+  openQasmGeneratedTsPath,
 );
+
+/**
+ * @param {any[]} sampleList
+ * @param {string} sampleDir
+ * @param {import("fs").PathOrFileDescriptor} generatedJsFileName
+ */
+function embedSampleContentsInTsFile(
+  sampleList,
+  sampleDir,
+  generatedJsFileName,
```

Comment:
Minor nit, but these are actually generated TS files, not JS files.

Reply:
I realized it late and fixed in some places, but not in all places.